os/fs/smartfs/smartfs_smart.c

```diff
@@ -245,20 +245,20 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 			goto errout_with_buffer;
 		}
 
-		/* TODO: Test open mode based on the file mode */
-
-		/* The file exists.  Check if we are opening it for O_CREAT or
-		 * O_TRUNC mode and delete the sector chain if we are. */
+		/* The file exists already.
+		 * If the file has been requested to be opened in TRUNCATE mode, then shrink the file.
```

Comment:
What about "If the file is opened in truncate mode, then shrink the file." ?

Reply:
Since both sentences mean the same and are understandable enough, I'll keep it same for now, thanks.